src/document/DocumentAnnotator.ts

```diff
@@ -152,4 +141,26 @@ export default class DocumentAnnotator extends BaseAnnotator {
             });
         }
     }
+
+    setupHighlightListener(pageEl: HTMLElement, textLayer: HTMLElement): void {
+        const pageNumber = this.getPageNumber(pageEl);
+        const prevHighlightListener = this.highlightListeners.get(pageNumber);
+
+        // Destroy any previous highlight listener since that element no longer exists
+        if (prevHighlightListener) {
+            prevHighlightListener.destroy();
+            this.highlightListeners.delete(pageNumber);
```

Comment:
Should this be folded into the "managers" concept to avoid having its own per-page collection? Can it become HighlightCreateManager or something similar?

Reply:
Yeah that's a good idea, I'll make that change